vulkan/vulkan_native_surface.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_VULKAN_VULKAN_NATIVE_SURFACE_H_
+#define FLUTTER_VULKAN_VULKAN_NATIVE_SURFACE_H_
+
+#include "lib/ftl/macros.h"
+#include "third_party/skia/include/core/SkSize.h"
+#include "vulkan_handle.h"
+#include "vulkan_proc_table.h"
+
+namespace vulkan {
+
+class VulkanNativeSurface {
```

Comment:
> Would this refactoring support adding a VulkanNativeSurfaceGLFW subclass?

Yes.

It seems to me that it would work. PlatformViewGLFW would no longer implement GPUSurfaceGLDelegate; if it were to follow the example of this CL, there would be new classes GLFWSurfaceGL and GLFWSurfaceVulkan, analogous to AndroidSurfaceGL and AndroidSurfaceVulkan. Is this correct?

Yes. One per client rendering API. Most of the Vulkan stuff is platform agnostic. But we still need the logic that takes a native window and converts it to a VkSurfaceKHR that has to be platform specific.